awsx/ec2/vpc.ts

```diff
@@ -343,14 +314,100 @@ export class Vpc extends schema.Vpc<VpcData> {
       routes,
       natGateways,
       eips,
-      subnetLayout: subnetLayout,
+      subnetLayout: rewrap(pulumi.output(subnetLayout)),
       privateSubnetIds,
       publicSubnetIds,
       isolatedSubnetIds,
       vpcId,
     };
   }
 
+  private decideSubnetSpecs(
+    name: string,
+    cidrBlock: pulumi.Input<string>,
+    subnetStrategy: schema.SubnetAllocationStrategyInputs,
+    availabilityZones: string[],
+    args: {
+      readonly subnetSpecs?: schema.SubnetSpecInputs[];
+      readonly subnetStrategy?: schema.SubnetAllocationStrategyInputs;
+      readonly availabilityZoneCidrMask?: number;
+    },
+  ): {
+    subnetSpecs: SubnetSpecPartial[];
+    subnetLayout: pulumi.Output<schema.ResolvedSubnetSpecOutputs[]>;
+  }{
+    const parsedSpecs = validateAndNormalizeSubnetInputs(
+      args.subnetSpecs,
+      availabilityZones.length,
+    );
+
+    const thisResource = this;
+
+    const subnetSpecs = (() => {
+
+      // cidrBlock is allocated dynamically but there are no explicit specs; use subnetDistributorNew.
```

Comment:
The case analysis here is a bit uncertain for me, appreciate some help formulating which test cases we want to add. 

One particular complication is that `getSubnetSpecsLegacy` cannot yet accept a `cidrBlock: Input<string>` and therefore this does not support IPAM-allocated cidrBlock with Legacy strategy properly. I've only done the Input threading for `getSubnetSpecs` which implements the newer Auto strategy.

Reply:
Would it be acceptable to only support IPAM when using the `Auto` strategy? Users won't be adding IPAM to an existing VPC, so I don't think we should have to support "legacy" stuff.

Reply:
It now only supports the Auto strategy. Indeed cleaner.